### DISCUSSION TITLE
does automatic prettier work as described in docs

### DISCUSSION THREAD

**Discussion by richtong** [LazyVim/LazyVim#4590] [Q&A]
I'm trying to use your very cool but I can't figure out how it actually works. I'm getting a "supported" is not definted which makes sense. i don't see where it is. Is there a typo somewher ih the docs? https://www.lazyvim.org/extras/formatting/prettier

Should the supported be something else I tried opts.supported and then gave up :-)

```
{ "stevearc/conform.nvim",
  opts = function(_, opts)
  opts.formatters_by_ft = opts.formatters_by_ft or {}
  for _, ft in ipairs(supported) do
    opts.formatters_by_ft[ft] = { "prettier" }
  end

  opts.formatters = opts.formatters or {}
  opts.formatters.prettier = {
    condition = function(_, ctx)
      return M.has_parser(ctx) and (vim.g.lazyvim_prettier_needs_config ~= true or M.has_config(ctx))
    end,
  }
end
}

```

**Comment by dpetka2001** [comment]
You don't need to copy what is mentioned on the LazyVim website to your own config. It's also mentioned in the docs
> caution
You don't need to copy the default settings to your config. They are only shown here for reference.

You just need to enable the Extra as mentioned in the docs
> info
You can enable the extra with the `:LazyExtras` command. Plugins marked as optional will only be configured if they are installed.

I don't use `prettier` myself, but I believe it should work if you just enable it from `:LazyExtras` command.
